### PR TITLE
create testimonials

### PR DESCRIPTION
**Pull request by Jula000**

### CODE REVIEW

File: src/app/data/testimonials.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    text: "Damien has an incredible talent for making every event feel effortless, and the results speak for themselves.",
+  },
+  {
+    name: "Samantha Davis",
+    location: "USA, California",
+    rating: 5,
+    text: "I was blown away by Damien's ability to capture the essence of our wedding day. His photographs are our cherished memories.",
+  },
+];
+export default testimonials;
```
Comment (yevhenii-horbenko) [inline src/app/data/testimonials.jsx]: for now I'd move it to Testimonials.jsx
also this file should've been named testimonials.js as it doesn't have a jsx syntax, but it doesn't matter now anyway

File: src/app/components/common/GenerateStars.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const generateStars = (rating) => {
+  let stars = [];
+  for (let i = 0; i < rating; i++) {
+    stars.push(
+      <span key={i} className="text-yellow-400">
+        ★
+      </span>
+    );
+  }
+  return stars;
```
Comment (yevhenii-horbenko) [inline src/app/components/common/GenerateStars.jsx]: you could just 
```
return (<>
{stars.map((_, index) => (
      <span key={index} className="text-yellow-400">
        ★
      </span>
)}
<>);
```

File: src/app/components/common/GenerateStars.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const generateStars = (rating) => {
```
Comment (yevhenii-horbenko) [inline src/app/components/common/GenerateStars.jsx]: I'd name this component TestimonialsRatings

**Review by yevhenii-horbenko — CHANGES_REQUESTED** [review]

File: src/app/components/common/TestimonialsRatings.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const TestimonialsRatings = ({ rating }) => {
```
Comment (yevhenii-horbenko) [inline src/app/components/common/TestimonialsRatings.jsx]: I'm not sure that this component is common

File: src/app/components/common/GlossyButton.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  const handleClick = (e) => {
+    console.log("Button was clicked");
+  };
+
+  return (
+    <Link href="/your-target-path" className="inline-block">
+      <button
+        onClick={handleClick}
+        className="bg-button-blue px-8 py-3 rounded-full ml-3 flex items-center justify-center shadow-lg"
+      >
```
Comment (yevhenii-horbenko) [inline src/app/components/common/GlossyButton.jsx]: can you use existing Button component?

File: src/app/components/common/GlossyButton.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+"use client";
+import Image from "next/image";
+import Link from "next/link";
+
+const GlossyButton = () => {
+  const handleClick = (e) => {
+    console.log("Button was clicked");
+  };
```
Comment (yevhenii-horbenko) [inline src/app/components/common/GlossyButton.jsx]: we probably don't need it

File: src/app/components/common/GlossyButton.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+"use client";
```
Comment (yevhenii-horbenko) [inline src/app/components/common/GlossyButton.jsx]: do you need use client here?

File: src/app/components/common/GlossyButton.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import Image from "next/image";
+import Link from "next/link";
+
+const GlossyButton = () => {
+  const handleClick = (e) => {
+    console.log("Button was clicked");
+  };
+
+  return (
+    <Link href="/your-target-path" className="inline-block">
```
Comment (yevhenii-horbenko) [inline src/app/components/common/GlossyButton.jsx]: this component should probably accept href from props

File: src/app/components/Testimonials.jsx [hunk 9]
```diff
@@ -1,12 +1,32 @@
 import Link from "next/link";
```
Comment (yevhenii-horbenko) [inline src/app/components/Testimonials.jsx]: I recommend creating a folder Testimonials
Create index.jsx file inside and move this content inside
And create folder Ratings inside Testimonials folder and move common/TestimonialsRatings.jsx there

So it'll look like:
/components/Testimonials/index.jsx
/components/Testimonials/Ratings/index.jsx

Comment (yevhenii-horbenko) [reply src/app/components/Testimonials.jsx]: This way you'll cleanup your common folder and Ratings will be isolated inside Testimonials folder and that's for good as we don't use it anywhere else anyway

**Review by yevhenii-horbenko — CHANGES_REQUESTED** [review]
great job Kate, I can see a huge progress
please check the comments that I've made

File: src/app/components/common/GlossyButton.jsx (hunk 7 above)
Comment (Jula000) [reply src/app/components/common/GlossyButton.jsx]: Screen Shot 2024-06-26 at 12 08 13 PM

File: src/app/components/testimonials/Testimonials.jsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
+import Link from "next/link";
+import Button from "../common/Button";
+import SocialMediaButtons from "../common/SocialMediaBtn";
+import TestimonialsRatings from "../testimonials/TestimonialsRatings";
```
Comment (yevhenii-horbenko) [inline src/app/components/testimonials/Testimonials.jsx]: here you exiting the folder and entering it again
I'd change it to
```suggestion
import Ratings from "./Ratings";
```

File: src/app/components/testimonials/TestimonialsRatings.jsx [hunk 11]
```diff
@@ -0,0 +1,15 @@
+const TestimonialsRatings = ({ rating }) => {
```
Comment (yevhenii-horbenko) [inline src/app/components/testimonials/TestimonialsRatings.jsx]: please check my comment from my previous review

File: src/app/page.tsx [hunk 12]
```diff
@@ -5,6 +5,7 @@ import AboutComponent from "./components/AboutComponent";
 import ServiceComponent from "./components/ServiceComponent";
 import Portfolio from "./components/Portfolio";
 import FaqComponent from "./components/FaqComponent";
+import Testimonials from "./components/testimonials/Testimonials";
```
Comment (yevhenii-horbenko) [inline src/app/page.tsx]:
```suggestion
import Testimonials from "./components/Testimonials";
```

**Review by yevhenii-horbenko — CHANGES_REQUESTED** [review]

**Review by yevhenii-horbenko — APPROVED** [review]
looks great!
ship it!
gj Kate

**Pull request merged**